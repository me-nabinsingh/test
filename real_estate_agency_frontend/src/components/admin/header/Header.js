import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Logo from '../../../components/logo/Logo';
import history from "../../../util/history";

class   Header extends Component {

    componentWillMount() {
        const token = localStorage.getItem('token');
        if(!token) {
            this._redirectToLogin();
        }
    }

    _redirectToLogin = () => {
        localStorage.clear();
        history.push('/auth/login');
    }

    _signOut = (e) => {
        e.preventDefault();
        this._redirectToLogin();
    }

    render() {
        return ( 
            <nav className="navbar navbar-dark fixed-top bg-dark flex-md-nowrap p-0 shadow">
                <div className="mx-5 my-2">
                    <Logo />
                </div>
                <ul className="navbar-nav px-3">
                    <li className="nav-item text-nowrap">
                        <Link to="#" className="nav-link" onClick={this._signOut}>Sign out</Link>
                    </li>
                </ul>
            </nav>
         );
    }
}
 
export default Header;