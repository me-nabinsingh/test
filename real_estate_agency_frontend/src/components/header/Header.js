import React from "react";
import { Link } from "react-router-dom";
import Logo from "../logo/Logo";
import  "./styles.css";

const Header = () => {
    const currentPath = window.location.pathname
    if(currentPath.includes('/admin') || currentPath.includes('/auth')) {
        return null;
    }
    return ( 
        <nav className="site-header sticky-top py-1">
            <div className="container d-flex flex-column flex-md-row justify-content-between">
                <Logo width={30} height={30} />
                <Link className="py-2 d-none d-md-inline-block" to="/admin/login">Admin</Link>
            </div>
        </nav>
    );
}
 
export default Header;