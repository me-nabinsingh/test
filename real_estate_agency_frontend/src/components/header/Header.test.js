import React from 'react';
import Header from './Header';
import {Router } from "react-router-dom";
import renderer from 'react-test-renderer';
import history from "../../util/history";


describe('<Header />', () => {
    it('renders without crashing', () => {
        const component = renderer.create(
            <Router history={history}>
                <Header/>
            </Router>
            );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});