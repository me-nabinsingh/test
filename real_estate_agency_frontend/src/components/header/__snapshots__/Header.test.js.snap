// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders without crashing 1`] = `
<nav
  className="site-header sticky-top py-1"
>
  <div
    className="container d-flex flex-column flex-md-row justify-content-between"
  >
    <a
      className="py-2"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        height={30}
        src="logo.svg"
        width={30}
      />
    </a>
    <a
      className="py-2 d-none d-md-inline-block"
      href="/admin/login"
      onClick={[Function]}
    >
      Admin
    </a>
  </div>
</nav>
`;
